rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/savedFilters.spec.ts

```diff
@@ -0,0 +1,198 @@
+import { mount, VueWrapper } from "@vue/test-utils";
+import SavedFilters from "../savedFilters.vue";
+import { ActivityFilterStore } from "../../../../library/stores/ActivityFilterStore";
+import { MessageBox } from "uiv";
+
+jest.mock("@/library/rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({ projectName: "test" }),
+}));
+
+jest.mock("uiv", () => ({
+  MessageBox: {
+    confirm: jest.fn().mockResolvedValue(true),
+    prompt: jest.fn().mockResolvedValue({ value: "test" }),
+  },
+  Notification: {
+    notify: jest.fn(),
+  },
+}));
+const originalFilters = [
+  { filterName: "Filter 1" },
+  { filterName: "Filter 2" },
+  { filterName: "Filter 3" },
+];
+
+jest.mock("../../../../library/stores/ActivityFilterStore", () => {
+  const originalModule = jest.requireActual(
+    "../../../../library/stores/ActivityFilterStore"
+  );
+  return {
+    __esModule: true,
+    ...originalModule,
+    ActivityFilterStore: jest.fn().mockImplementation(() => {
+      return {
+        filters: [...originalFilters],
+        loadForProject: jest.fn().mockImplementation(() => {
+          return { filters: [...originalFilters] };
+        }),
+        removeFilter: jest
+          .fn()
+          .mockImplementation((projectName: string, filterName: string) => {
+            const index = originalFilters.findIndex(
+              (filter) => filter.filterName === filterName
+            );
+            if (index !== -1) {
+              originalFilters.splice(index, 1);
+            }
+            return Promise.resolve();
+          }),
+        saveFilter: jest.fn().mockResolvedValue({}),
+      };
+    }),
+  };
+});
+
+const mountSavedFilters = (
+  props = {},
+  filterStoreOverrides = {}
+): VueWrapper<any> => {
+  const filterStore = {
+    ...new ActivityFilterStore(),
+    ...filterStoreOverrides,
+  };
+  return mount(SavedFilters, {
+    props: {
+      hasQuery: true,
+      query: {},
+      eventBus: {
+        on: jest.fn(),
+        emit: jest.fn(),
+        off: jest.fn(),
+        all: new Map(),
+      },
+      ...props,
+    },
+    global: {
+      mocks: {
+        $t: (msg) => msg,
+        filterStore,
```

Comment:
to do:
- add this `$confirm: jest.fn().mockResolvedValue(true),` after filterStore on line 78